spring-integration-core/src/main/java/org/springframework/integration/config/IntegrationRegistrar.java

```diff
@@ -220,6 +220,17 @@ private void registerBuiltInBeans(BeanDefinitionRegistry registry) {
 						"there is no jayway json-path.jar on the classpath.");
 			}
 
+			try {
+				ClassUtils.forName("com.jayway.jsonpath.Predicate", this.classLoader);
+			}
+			catch (ClassNotFoundException e) {
+				jsonPathClass = null;
+				logger.error("SpEL function '#jsonPath' isn't registered. " +
```

Comment:
No; I mean if we've already detected that json path is __not__ on the CP we shouldn't even do this test. It should be in the 

    if (jsonPathClass != null) {
        ...
    }

below